### ISSUE TITLE
torchtext version issue

### ISSUE THREAD

**Issue by CarreraPHP** [apple/corenet#14]
@team,

my Mac Specification - Mac M3 Max 128GB RAM and 4TB Storage.

I cloned the repo and trying to install below error occurs

```Shell
ERROR: Could not find a version that satisfies the requirement torchtext==0.17.1 (from corenet) (from versions: 0.1.1, 0.2.0, 0.2.1, 0.2.3, 0.3.1, 0.4.0, 0.5.0, 0.6.0, 0.16.2, 0.17.2, 0.18.0)
ERROR: No matching distribution found for torchtext==0.17.1
```

I am able to get the torchtext 0.18.0 but again it is not working.

**Comment by CarreraPHP** [comment]
I use conda for python and installed latest python version

**Comment by rezashahnazar** [comment]
I initially had the same issue on MacBook Pro M1; However, it fixed when I explicitly mentioned the python version 11 (instead of default 12 on my system) when installing venv by changing the following part of installation:
```
python3.11 -m venv venv && source venv/bin/activate
```
I think its reasonable to have an improvement on the documentation around versions.

**Comment by ruslan1310** [comment]
hello

**Comment by michaelrrrrrr** [comment]
> I initially had the same issue on MacBook Pro M1; However, it fixed when I explicitly mentioned the python version 11 (instead of default 12 on my system) when installing venv by changing the following part of installation:
> 
> ```
> python3.11 -m venv venv && source venv/bin/activate
> ```
> 
> I think its reasonable to have an improvement on the documentation around versions.

After changing Python version to 3.11.9 on Ubuntu Desktop 24.04, everything works fine.

**Comment by mohammad7t** [comment]
Hi @CarreraPHP  @rezashahnazar and @michaelrrrrrr,

Thank you for bringing this into our attention and thank you for your suggestions. My understanding from your suggestions is that the `torchtext==0.17.1` dependency fails to get installed on python-3.12. We will look into this and update our requirements and/or documentation.

**Comment by mohammad7t** [comment]
Hi all,
We've just merged #37 , which should fix this issue. I'm closing this issue, but please feel free to re-open it if you face issues with `torchtext==0.17.2`
